### PR TITLE
EVG-7826 generate model conversion methods

### PR DESCRIPTION
**Pull request by john-m-liu**
Sorry this is more WIP-y than usual. This adds very bare-bones functionality to extract fields and types from a db model and generate the conversion methods for the rest model. Does not handle many types yet and is not connected to the generation of the model itself (that's a later ticket). The test will also be redone to be in the same pattern as the model generator test

### CODE REVIEW

File: rest/model/codegen.go [hunk 1]
```diff
@@ -176,3 +191,89 @@ func goimports(source string) ([]byte, error) {
 		AllErrors: true, Comments: true, TabIndent: true, TabWidth: 8,
 	})
 }
+
+func CreateConversionMethods(packageName, structName string, fields ExtractedFields) ([]byte, error) {
+	pkg, err := importer.Default().Import(packageName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to resolve package '%s'", packageName)
+	}
+	scope := pkg.Scope()
```
Comment (ablack12) [inline rest/model/codegen.go:200]: Is it possible for scope to be nil? (if not you don't use this anywhere else so you can just combine it with line 201)

File: rest/model/conversions.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+func intPtrToInt(in *int) int {
+	if in == nil {
+		return 0
+	}
+	return *in
+}
+
+func intPtrToIntPtr(in int) *int {
```
Comment (ablack12) [inline rest/model/conversions.go:45]: This function should take `in *int` and return in, I believe

File: rest/model/codegen.go (hunk 1 above)
Comment (john-m-liu) [reply rest/model/codegen.go:200]: added handling for nil

File: rest/model/conversions.go (hunk 2 above)
Comment (john-m-liu) [reply rest/model/conversions.go]: oops fixed

**Pull request merged**